src/extension/mapml/src/test/java/org/geoserver/mapml/MapMLDimensionsTest.java

```diff
@@ -177,24 +159,192 @@ public void testTimeList() throws Exception {
+        ResourceInfo layerMeta = catalog.getLayerByName(getLayerId(WATTEMP)).getResource();
+        assertTrue(layerMeta instanceof CoverageInfo);
+        CoverageInfo coverageInfo = (CoverageInfo) layerMeta;
+        // layer has no dimension yet
+        assertNull(coverageInfo.getMetadata().get(ResourceInfo.ELEVATION, DimensionInfo.class));
+        setupRasterDimension(
+                WATTEMP, ResourceInfo.ELEVATION, DimensionPresentation.LIST, null, null, null);
+
+        // re-query the catalog for the updated info
+        coverageInfo = (CoverageInfo) catalog.getLayerByName(getLayerId(WATTEMP)).getResource();
+        // get the dimension info from the metadata map
+        DimensionInfo elevationInfo =
+                coverageInfo.getMetadata().get(ResourceInfo.ELEVATION, DimensionInfo.class);
+        // prove it's enabled, but not yet known to mapml
+        assertTrue(elevationInfo.isEnabled());
+
+        // get the mapml doc for the layer
+        String path =
+                "wms?LAYERS="
+                        + getLayerId(WATTEMP)
+                        + "&STYLES=&FORMAT="
+                        + MapMLConstants.MAPML_MIME_TYPE
+                        + "&SERVICE=WMS&VERSION=1.3.0"
+                        + "&REQUEST=GetMap"
+                        + "&SRS=EPSG:3857"
+                        + "&BBOX=0,0,1,1"
+                        + "&WIDTH=150"
+                        + "&HEIGHT=150"
+                        + "&format_options="
+                        + MapMLConstants.MAPML_WMS_MIME_TYPE_OPTION
+                        + ":image/png";
+        org.w3c.dom.Document doc = getMapML(path);
+        // assure us it's actually working as a document
+        assertXpathEvaluatesTo("1", "count(//html:map-link[@rel='image'][@tref])", doc);
+        URL url = new URL(xpath.evaluate("//html:map-link[@rel='image']/@tref", doc));
+        // assert that there's no time variable nor URL template variable
+        HashMap<String, String> vars = parseQuery(url);
+        assertNull(vars.get("time"));
+        assertXpathEvaluatesTo("0", "count(//html:map-select[@name='elevation'])", doc);
+        // update the layer metadata to indicate the dimension is known to mapml
+        // doesn't test the web interface, but the does test the mechanism
+        // the web interface uses to signal dimension for mapml
+        layerMeta = catalog.getLayerByName(getLayerId(WATTEMP)).getResource();
+        assertMapMlDocumentHasDimension(catalog, layerMeta, path, "elevation", "2");
+    }
+
+    private void assertMapMlDocumentHasDimension(
+            Catalog catalog,
+            ResourceInfo layerMeta,
+            String path,
+            String dimension,
+            String expectedCount)
+            throws Exception {
         MetadataMap mm = layerMeta.getMetadata();
-        mm.put("mapml.dimension", "time");
+        mm.put("mapml.dimension", dimension);
         catalog.save(layerMeta);
```

Comment:
I find it confusing that the MapML dimension configuration is part of an assert that checks if the output has the dimension. Should be part of the other setup you have before calling this method, IMHO, eventually in its own method if you want to re-use code.

Reply:
Feedback addressed, dedicated method has been created and moved out from within the assertion.